.github/workflows/release-proxy-init.yml

```diff
@@ -0,0 +1,96 @@
+name: Proxy-init release
+
+on:
+  pull_request:
+    paths:
+      - .github/workflows/release-proxy-init.yml
+  push:
+    tags: ["init/v*"]
+
+permissions:
+  contents: read
+
+jobs:
+  meta:
+    timeout-minutes: 3
+    runs-on: ubuntu-latest
+    container: ghcr.io/linkerd/dev:v30-tools
+    steps:
+      - id: meta
+        shell: bash
+        run: |
+          set -eu
+          shopt -s extglob
+          ref='${{ github.ref }}'
+          if [[ "$ref" == refs/tags/proxy-init/v+([0-9]).+([0-9]).+([0-9])?(-+([a-z0-9-])) ]]; then
+            echo ::set-output name=version::"${ref##refs/tags/proxy-init/}"
+            echo ::set-output name=mode::release
+          else
+            sha='${{ github.sha }}'
+            echo ::set-output name=version::"test-${sha:0:7}"
+            echo ::set-output name=mode::test
+          fi
+    outputs:
+      repo: ghcr.io/${{ github.repository_owner }}/proxy-init
+      mode: ${{ steps.meta.outputs.mode }}
+      version: ${{ steps.meta.outputs.version }}
```

Comment:
nit, take it or leave it: would it be worth factoring this out into a bash script/action that can just be run with the expected regex? not a big deal...